components/enterprise/subscriptions.tf

```diff
@@ -1,62 +1,37 @@
-  cft_non_production = { for k, v in var.cft_non_production_subscriptions : k => merge(tomap({ group = "cft_non_production" }), v) }
-  cft_production     = { for k, v in var.cft_production_subscriptions : k => merge(tomap({ group = "cft_production" }), v) }
-
-  sds                = { for k, v in var.sds_subscriptions : k => merge(tomap({ group = "sds" }), v) }
-  sds_sandbox        = { for k, v in var.sds_sandbox_subscriptions : k => merge(tomap({ group = "sds_sandbox" }), v) }
-  sds_non_production = { for k, v in var.sds_non_production_subscriptions : k => merge(tomap({ group = "sds_non_production" }), v) }
-  sds_production     = { for k, v in var.sds_production_subscriptions : k => merge(tomap({ group = "sds_production" }), v) }
-
-  crime = { for k, v in var.crime_subscriptions : k => merge(tomap({ group = "crime" }), v) }
-
-  heritage                = { for k, v in var.heritage_subscriptions : k => merge(tomap({ group = "heritage" }), v) }
-  heritage_sandbox        = { for k, v in var.heritage_sandbox_subscriptions : k => merge(tomap({ group = "heritage_sandbox" }), v) }
-  heritage_non_production = { for k, v in var.heritage_non_production_subscriptions : k => merge(tomap({ group = "heritage_non_production" }), v) }
-  heritage_production     = { for k, v in var.heritage_production_subscriptions : k => merge(tomap({ group = "heritage_production" }), v) }
-
-  security = { for k, v in var.security_subscriptions : k => merge(tomap({ group = "security" }), v) }
-
-  platform                = { for k, v in var.platform_subscriptions : k => merge(tomap({ group = "platform" }), v) }
-  platform_sandbox        = { for k, v in var.platform_sandbox_subscriptions : k => merge(tomap({ group = "platform_sandbox" }), v) }
-  platform_non_production = { for k, v in var.platform_non_production_subscriptions : k => merge(tomap({ group = "platform_non_production" }), v) }
-  platform_production     = { for k, v in var.platform_production_subscriptions : k => merge(tomap({ group = "platform_production" }), v) }
-
-  subscriptions = merge(
-    local.cft,
-    local.cft_sandbox,
-    local.cft_non_production,
-    local.cft_production,
-
-    local.sds,
-    local.sds_sandbox,
-    local.sds_non_production,
-    local.sds_production,
-
-    local.crime,
-
-    local.heritage,
-    local.heritage_sandbox,
-    local.heritage_non_production,
-    local.heritage_production,
-
-    local.security,
-
-    local.platform,
-    local.platform_sandbox,
-    local.platform_non_production,
-    local.platform_production,
-  )
-}
-
 module "subscription" {
   for_each = local.subscriptions
 
-  source = "../../modules/subscription"
-  name   = each.key
-  value  = each.value
+  source      = "../../modules/subscription"
+  name        = each.key
+  value       = each.value
+  common_tags = module.tags[each.key].common_tags
+  environment = try(each.value.environment, (lower(replace([each.key][0], "/.*-([A-Za-z]+).*/", "$1"))))
```

Comment:
we should probably use a regex extracting known environments and test that it fails if it doesn't match, i.e. we don't want people using NLE

Reply:
How about this? https://github.com/hmcts/azure-enterprise/pull/9/files#diff-c2d73feb59fa40481d75713d380ce32b59d7cb817246b40ddda5909a0068a143R57-R64

Reply:
yeah fine